### PR TITLE
Simplify dev/test environment deployment

### PR DESCRIPTION
**Pull request by gl-johnson**
### Desired Outcome

Clean up some redundancies in the deployment scripts, and make it simpler to specify which mode to run Secrets Provider in.

### Implemented Changes

- Consolidated deploy_init_env, deploy_k8s_rotation_env, and deploy_push_to_file into one function
- Update test cases and dev bootstrap.env to use `SECRETS_MODE` to set the mode for deployment

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml [hunk 1]
```diff
@@ -67,16 +67,16 @@ spec:
...
-      - image: '${PULL_DOCKER_REGISTRY_PATH}/${APP_NAMESPACE_NAME}/debian:latest'
+      - image: debian
         name: test-app
         command: ["sleep"]
         args: ["infinity"]
         volumeMounts:
           - mountPath: /opt/secrets/conjur
             name: conjur-secrets
             readOnly: true
-      - image: '${PULL_DOCKER_REGISTRY_PATH}/${APP_NAMESPACE_NAME}/secrets-provider:latest'
```
[the remark this answers is not in the document]
Comment (gl-johnson) [reply deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml:78]: These manifests are only used when running `./bin/start` with the `--dev` flag, so I think we can assume that the image already exists locally per the docs to [deploy a local dev environment](https://github.com/cyberark/secrets-provider-for-k8s/blob/main/CONTRIBUTING.md#deploy-a-local-development-environment), the first step of which is to run `./bin/build`.

File: deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml [hunk 2]
```diff
@@ -67,16 +67,16 @@ spec:
...
-        imagePullPolicy: Always
+      - image: 'secrets-provider-for-k8s:latest'
+        imagePullPolicy: Never
```
[the remark this answers is not in the document]
Comment (gl-johnson) [reply deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml:79]: Hmm I'm not sure on this. This change is largely based on the existing dev manifests for K8s secrets mode, both of which omitted the registry path variables and use an imagePullPolicy of Never.

I think the assumption once again is that `./bin/build` was run prior to the start script, which tags the locally built secrets-provider image as `latest`

[the remark this answers is not in the document]
Comment (rpothier) [reply deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml:79]: I guess my concern is if I rebuild a dev image with a small change, and redeploy the pod will the latest build be picked up?

File: deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpothier) [reply deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml:78]: OK, if it's only the dev image that's fine.

File: deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gl-johnson) [reply deploy/dev/config/k8s/secrets-provider-p2f-rotation.sh.yml:79]: Yes it should still be picked up as far as I can tell. Each time you run `./bin/build` it overwrites dev/latest tags for the image locally, which is what will be used in the deployment when the imagePullPolicy is 'Never'.

I tested it with `./bin/start --dev` and `./bin/start --dev --reload` and it picks up code changes

**Pull request merged**